cmd/ingest/main.go

```diff
@@ -90,107 +93,107 @@ func main() {
 	if err != nil {
 		log.WithFields(logrus.Fields{
 			"error": err,
-		}).Fatal("couldn't create zmq context")
-	}
-	defer ctx.Term()
+		}).Warn("zcash.conf failed, will try empty credentials for rpc")
 
-	// WARNING: The Socket is not thread safe. This means that you cannot
-	// access the same Socket from different goroutines without using something
-	// like a mutex.
-	sock, err := ctx.NewSocket(zmq.SUB)
-	if err != nil {
-		log.WithFields(logrus.Fields{
-			"error": err,
-		}).Fatal("couldn't create zmq context socket")
+		//Default to testnet, but user MUST specify rpcuser and rpcpassword in zcash.conf; no default
+		rpcClient, err = frontend.NewZRPCFromCreds("127.0.0.1:18232", " ", " ")
+
+		if err != nil {
+			log.WithFields(logrus.Fields{
+				"error": err,
+			}).Fatal("couldn't start rpc connection")
+		}
 	}
 
-	err = sock.SetSubscribe(opts.zmqTopic)
+	ctx := context.Background()
+	height, err := storage.GetCurrentHeight(ctx, db)
 	if err != nil {
 		log.WithFields(logrus.Fields{
-			"error":  err,
-			"stream": opts.zmqTopic,
-		}).Fatal("couldn't subscribe to stream")
+			"error": err,
+  	}).Warn("unable to get current height from local db storage")
+		
 	}
-
-	connString := fmt.Sprintf("tcp://%s", opts.zmqAddr)
-
-	err = sock.Connect(connString)
-	if err != nil {
+	
+	if height < 0 {
+		height = 0
 		log.WithFields(logrus.Fields{
-			"error":      err,
-			"connection": connString,
-		}).Fatal("couldn't connect to socket")
+			"error": err,
+  	}).Warn("invalid current height read from local db storage")
 	}
-	defer sock.Close()
-
-	log.Printf("Listening to 0mq on %s", opts.zmqAddr)
-
+	 
 	// Start listening for new blocks
 	for {
-		msg, err := sock.RecvMessageBytes(0)
-		if err != nil {
+		block, err := getBlock(rpcClient, height)
+		if err != nil{
 			log.WithFields(logrus.Fields{
-				"error": err,
-			}).Error("error on msg recv")
-			continue
+				"height": height,
+				"error":  err,
+			}).Fatal("error with getblock")
 		}
-
-		if len(msg) < 3 {
-			log.WithFields(logrus.Fields{
-				"msg": msg,
-			}).Warn("got unknown message type")
-			continue
+		if block != nil{
+			handleBlock(db , block)
+			height++
+			//TODO store block current/prev hash for formal reorg
+		}else{
+			//TODO implement blocknotify to minimize polling on corner cases
+			time.Sleep(60 * time.Second)
 		}
+	}
+}
 
-		topic, body := msg[0], msg[1]
-
-		var sequence int
-		if len(msg[2]) == 4 {
-			sequence = int(binary.LittleEndian.Uint32(msg[len(msg)-1]))
+func getBlock(rpcClient *rpcclient.Client, height int) (*parser.Block, error) {
+	params := make([]json.RawMessage, 2)
+	params[0] = json.RawMessage("\"" + strconv.Itoa(height) + "\"")
+	params[1] = json.RawMessage("0")
+	result, rpcErr := rpcClient.RawRequest("getblock", params)
+
+	var err error
+	var errCode int64
+
+	// For some reason, the error responses are not JSON
+	if rpcErr != nil {
+		errParts := strings.SplitN(rpcErr.Error(), ":", 2)
```

Comment:
Add a comment saying why we are checking for error code `-8` (namely that it indicates we don't have the requested block height yet).